src/vcpkg/registries.cpp

```diff
@@ -491,101 +532,95 @@ namespace
     // { GitRegistry::RegistryImplementation
     std::unique_ptr<RegistryEntry> GitRegistry::get_port_entry(const VcpkgPaths& paths, StringView port_name) const
     {
-        auto port_versions = get_versions_tree_path(paths);
-        auto maybe_version_entries =
-            load_versions_file(paths.get_filesystem(), VersionDbType::Git, port_versions, port_name);
-        Checks::check_maybe_upgrade(
-            VCPKG_LINE_INFO, maybe_version_entries.has_value(), "Error: " + maybe_version_entries.error());
-        auto version_entries = std::move(maybe_version_entries).value_or_exit(VCPKG_LINE_INFO);
+        if (!m_paths) m_paths = &paths;
+        return std::make_unique<GitRegistryEntry>(*this, port_name);
+    }
 
-        auto res = std::make_unique<GitRegistryEntry>();
-        res->port_name = port_name.to_string();
-        for (auto&& version_entry : version_entries)
-        {
-            res->port_versions.push_back(version_entry.version);
-            res->git_trees.push_back(version_entry.git_tree);
-        }
-        return res;
+    GitRegistryEntry::GitRegistryEntry(const GitRegistry& reg, StringView name)
+        : port_name(name.to_string()), parent(reg)
```

Comment:
Note to self: rip this `parent` bit out later.

Reply:
In general, this will not be possible -- shared state is mandated by the actual desired execution (if one GitRegistryEntry forces a `git fetch`, those results should apply to all other Entries). Forcing that shared state to be passed in to every API call results in a completely untestable interface, since the API is type-erased and as such every caller has to pass the superset of all possible shared states (VcpkgPaths) in.

It might be possible to reduce the "scope" of the captured parent pointer, by capturing several pointers to smaller subobjects, but fundamentally there is shared state here between the GitRegistryEntry objects and calls upon one affect the behavior of the others.